[android] Add link to Google Play TTS in voice settings

gpesquero opened the pull request:
Add of link to Google Play TTS website in "Voice Instructions" Settings menu.

The previous implementation shows the text of the link, but it's not an operational link:
![en_1](https://github.com/organicmaps/organicmaps/assets/25888296/afac34b5-7e33-45b1-b46f-ee72734093e0)

The new implementation adds a working link that directs to TTS Google Play website:
![en_2](https://github.com/organicmaps/organicmaps/assets/25888296/c6356a09-c861-4bc8-a5e6-5c314780ea84)

This PR also adds a description of the meaning of the TTS acronym (Text-To-Speech).

biodranik reviewed (commented):
1. Can links be clickable in the text?
2. Can words be clickable like links in a browser (without the visible link on the screen)?
3. Many Fdroid or Github apk users won't install Google TTS. So it's better to link several of them, maybe including FDroid links for FDroid builds. Or even better, we need to finish the updated instructions page https://github.com/organicmaps/organicmaps.github.io/pull/32 and reference all links there instead of complicating the UI. And leave only one link to the guide.

rtsisyk commented:
>  3. Many Fdroid or Github apk users won't install Google TTS. So it's better to link several of them, maybe including FDroid links for FDroid builds. Or even better, we need to finish the updated instructions page [[WIP]Added text to speech instructions organicmaps.github.io#32](https://github.com/organicmaps/organicmaps.github.io/pull/32) and reference all links there instead of complicating the UI. And leave only one link to the guide.

Let's focus on broken link here.

meenbeese commented:
Please also mention that this closes #884 @gpesquero

gpesquero commented:
> 1. Can links be clickable in the text?

Yes, links can be clickable in the text. I've implemented some piece of code to enable visible links in the text that can be clicked to open a web browser to the guide. In order to get links working, I've had to add a new `CustomPreference` class for the access to the `TextView` inside of `Preference`. This implementation may not be the best approach, but it works. Any feedback is welcomed.

> 2. Can words be clickable like links in a browser (without the visible link on the screen)?

Maybe words with no visible links can be clickable, but I don't see the point to have hidden links in the UI.

>3. Many Fdroid or Github apk users won't install Google TTS. So it's better to link several of them, maybe including FDroid links for FDroid builds. Or even better, we need to finish the updated instructions page [WIP]Added text to speech instructions organicmaps.github.io#32 and reference all links there instead of complicating the UI. And leave only one link to the guide.

I've modified the text so it does not include the link to Google TTS. The text now  is a mixture of the previous information messages, and it does not change depending if the TTS is enabled or not. The link still points to the previous "Maps.me" guide (to be modified once we get the new instructions page updated).

![en_3](https://github.com/organicmaps/organicmaps/assets/25888296/68d74c38-866c-4fa3-a42c-554ab13a697a)

biodranik commented:
1. It makes sense to merge this PR with the updated website link, after finishing/fixing https://github.com/organicmaps/organicmaps.github.io/pull/185 (any help is appreciated!)
2. Texts should be reworded for clarity. E.g. "we use..." => "Organic Maps uses...", "For some languages" => TTS is needed for any language, right?, etc.
3. Maybe texts should be customized for build flavors, I can imagine Google or Huawei not approving the app on review for mentioning an alternate app store.
4. Direct links to some synthesizers may be also inserted. Although a link for a guide with details is good.

gpesquero commented:
1. I've modified the link so it directs to the new URL, which I presume that it's going to be `https://organicmaps.app/faq/tts`.
2. I've reworded these instructions. Please check figure down below...
3. As the new TTS FAQ page seems to have quite a lot of detailed instructions for the different _flavours_, I've removed any reference here to any app store.
4. Again, as the new TTS FAQ page include links to many possible synthesizers, I've just left here the link to the new TTS FAQ page.

![en_04](https://github.com/organicmaps/organicmaps/assets/25888296/53935c2e-01e4-4b1e-bf2e-a14a21f11636)

Jean-BaptisteC reviewed (approved):
Pixel 6 - Android 14
Title of paragraph is it really necessary?

biodranik commented:
1. Checks are failing.
2. Title is not needed.
3. Wording should be improved to be more precise and succinct.

gpesquero commented:
1. Included `[strings] Regenerated` commit to avoid checks failure.
2. Removed title.
3. Removed second paragraph of the wording... Let's hope that it's succinct enough... ;) If not, please provide your proposal.

![en_05](https://github.com/organicmaps/organicmaps/assets/25888296/070a8b91-76e2-4175-a57f-2f107644f668)

gpesquero commented:
@biodranik: While we're dealing with [[WIP] F.A.Q + TTS #185](https://github.com/organicmaps/organicmaps.github.io/pull/185), could we also proceed with the merge of these other PRs related to TTS settings?...

1. https://github.com/organicmaps/organicmaps/pull/5974
2. https://github.com/organicmaps/organicmaps/pull/6125
3. https://github.com/organicmaps/organicmaps/pull/6142

Jean-BaptisteC commented:
Can you add in your first comment `closes #2383`

Jean-BaptisteC commented on android/app/src/main/java/app/organicmaps/settings/SettingsPrefsFragment.java:
```diff
@@ -373,23 +366,37 @@ else if (preference.getKey() != null && preference.getKey().equals(getString(R.s
+
+                              final String ttsFaqUrl = requireActivity().getString(R.string.tts_faq_url);
+
+                              final Intent intent = new Intent(Intent.ACTION_VIEW);
+                              intent.setData(Uri.parse(ttsFaqUrl));
+                              requireContext().startActivity(intent);
```
```suggestion
                              Utils.openUri(requireContext(), Uri.parse(ttsFaqUrl));
```

gpesquero replied:
Thanks @Jean-BaptisteC ... I will amend the commit...
